### PR TITLE
Api feature3

### PR DESCRIPTION
**Pull request by D-moe**
Add ability to upload images in JavaScript to comments section, currently deployed.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -110,9 +110,18 @@ function loadComments(path) {
...
-              let userName = json[index]['userName'];
-              let commentBody = json[index]['commentBody'];
-              addComment(/* name= */ userName, /* content= */ commentBody);
+              let currentIndex = json[index];
+              let userName = currentIndex['userName'];
+              let commentBody = currentIndex['commentBody'];
+              let imageUrl = undefined;
+              // we need to check whether an image Url exists
+              console.log(json);
+              if (json[index].hasOwnProperty('imageLink')) {
```
Comment (jmchristen) [inline portfolio/src/main/webapp/script.js:119]: You can replace `json[index]` with your variable.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -118,12 +133,55 @@ private void refreshData(HttpServletRequest request,
...
+
+  private String getUploadedFileUrl(HttpServletRequest request,
+                                    String formInputElementName) {
+    BlobstoreService blobstoreService =
+        BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev
+    // server)
```
[the remark this answers is not in the document]
Comment (D-moe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:154]: this was done in the example code, i followed there example, in the  implementation that ended up working on the server this was removed

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -282,4 +282,9 @@ li {
 #file-upload {
   display: inline-block;
   vertical-align: top;
+}
+
+.comment-full img{
+  width:10%;
+  height:auto;
```
Comment (jmchristen) [inline portfolio/src/main/webapp/style.css:289]: Space after colon.

File: portfolio/src/main/java/com/google/sps/objects/User.java [hunk 4]
```diff
@@ -5,15 +5,17 @@
...
+  public User(String userName, String commentBody, String imageLink) {
     this.userName = userName;
     this.commentBody = commentBody;
+    this.imageLink = imageLink;
   }
 
   public String getUserName() { return this.userName; }
-
   public String getCommentBody() { return this.commentBody; }
+  public String getImageLink() { return this.imageLink; }
```
Comment (jmchristen) [inline portfolio/src/main/java/com/google/sps/objects/User.java:18]: Blank line between methods.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -110,9 +110,18 @@ function loadComments(path) {
...
           json => {
             for (index in json) {
-              let userName = json[index]['userName'];
-              let commentBody = json[index]['commentBody'];
-              addComment(/* name= */ userName, /* content= */ commentBody);
+              let currentComment = json[index];
+              let userName = currentComment['userName'];
+              let commentBody = currentComment['commentBody'];
+              let imageUrl = undefined;
+              // we need to check whether an image Url exists
```
Comment (zjwoody) [inline portfolio/src/main/webapp/script.js:117]: Comment should end with period.

**Pull request merged**